content/blogposts/2022/sourcegraph-4.0.md

```diff
@@ -0,0 +1,197 @@
+  caption="High-level aggregations of search results"
+/>
+
+  <a href="https://docs.sourcegraph.com/code_insights/explanations/search_results_aggregations" className="tw-not-italic tw-flex tw-items-center">
+      <p className="tw-m-0 tw-mr-2">Docs</p>
+      <OpenInNewIcon className="tw-inline" size={18} />
+  </a>
+
+
+#### Configure precise code navigation for 9 languages in a matter of minutes with auto-indexing
+Code navigation allows you to quickly jump to symbol definitions, find references, dependencies, and more, so you can spend more time shipping code. Precise code navigation is the most powerful version of code navigation and it works cross-repository. Auto-indexing allows you to set up precise code navigation for any chosen repositories with the click of a button, and it makes precise code navigation work across all of your code, automatically. Auto-indexing is supported for 9 languages including Ruby, Rust, Go, Java, Scala, Kotlin, Python, TypeScript, JavaScript, and it is turned on by default for our Cloud customers.
+
+<Video 
+  source={{
+    webm: 'blog/release-post/4.0/ruby-precise',
+    mp4: 'blog/release-post/4.0/ruby-precise'
+  }}
+  loop={true}
+  title="Ruby precise code navigation"
+  caption="Precise code navigation with Ruby"
+/>
+
+[Docs](https://docs.sourcegraph.com/code_navigation/references/indexers)
+
+<Badge text="Extensions" color="vermillion" size="small" onClick={() => window.open("https://docs.sourcegraph.com/integration", "_self")} />
+
+#### Your favorite extensions are now available by default 
+we’re investing in a new model for integrations that will support deeper integrations with our code intelligence platform and make it easier to surface code context during the ideal moments in your workflow. Top used extensions, including code navigation, git-extras, open-in-editor, and search-exports, are now part of the core functionality. 
+
+By default, you won't be able to access the extensions registry to use or enable extensions, and you will no longer be able to create extensions on a private registry. Note: if you still need access to extensions or create extensions on a private registry, you can enable a feature flag to do so until early 2023.
+
+Please note that this *does not* impact our IDE extensions, which will continue to allow you to search and navigate across all of your repositories without ever leaving your IDE or checking them out locally
+
+
+![Git Blame](https://storage.googleapis.com/sourcegraph-assets/blog/release-post/4.0/git_blame.png "Git Blame")
+
+<Badge text="Code navigation" color="lemon" size="small" onClick={() => window.open("/code-search, "_self")} />
+
+#### Quickly access answers within your codebase with a revamped reference panel 
+We've enhanced the code navigation experience with a new version of the reference panel. Updates include a new preview panel, the conslidation of definitions and references in the same tab, and allows for easier navigation through filters.
```

Comment:
Asking @macraig on this since I'm unaware. Good points.

Reply:
- What is new about the preview panel - and how it helps
We didn't have a preview panel before, so every time you clicked on a reference the user would be immediately taken to the reference's file. Now you can navigate references with the preview panel without leaving the file you were originally looking at. You can choose to jump to a reference, but it's optional. Most of the time you're navigating code you just want to take a peek, the preview panel makes that possible without leaving the current file location.

- What it means to consolidate definitions and references - and why this matters
Before, definitions and references were located in different tabs. As the number of definitions is typically 1, it also felt like a bit of unused space. We now present both of them (and implementations) in the same tab by stacking them, which reduces the number of tabs needed and also promotes ease of navigation (scrolling instead of clicking tabs).

- What's new about the filters that make it easier to navigate
We didn't have any form of filtering in the old reference panel. This meant that you had to scroll down to find the reference you were looking for, which didn't scale for symbols that had a large number of results. Users can now easily filter results by file name and find the references they are looking for much faster.